packages/design-system/src/duration.ts

```diff
@@ -0,0 +1,124 @@
+import {
+    DurationUnit,
+    FullDate,
+    diffDates,
+    formatPresets,
+    getNowInUserTimezone,
+    isDateAfter,
+    toLocaleString,
+} from 'date-vir';
+
+export type HumanizeDurationUnit = 'month' | 'week' | 'day' | 'hour' | 'minute';
+
+export function formatFullDate(fullDate: FullDate): {
+    date: string;
+    time: string;
+} {
+    /**
+     * Note that the below .replace strings are to fix weird behavior in Chromium where the "NARROW
+     * NO-BREAK SPACE" character (code 8239) is used instead of a space (code 32) before AM/PM.
+     */
+    return {
+        date: toLocaleString(fullDate, {dateStyle: 'medium'}).replace(/\s/g, ' '),
+        time: toLocaleString(fullDate, {timeStyle: 'short'}).replace(/\s/g, ' '),
+    };
+}
+
+export function humanizeDuration({
+    date,
+    allowedRelativeUnits = [
+        'month',
+        'week',
+        'day',
+        'hour',
+        'minute',
+    ],
+}: {
+    date: FullDate;
+    /**
+     * Controls which relative durations are allowed. Always uses the lowest possible unit. IE, if
+     * 'week' is included, "X weeks ago" or "in X weeks" may be used (if the given date further ago
+     * than all other provided units). See tests for more examples.
+     */
+    allowedRelativeUnits?: HumanizeDurationUnit[] | undefined;
+}): string {
+    const {days} = diffDates({
+        start: date,
+        end: getNowInUserTimezone(),
+        unit: DurationUnit.Days,
+    });
+
+    const {hours} = diffDates({
+        start: date,
+        end: getNowInUserTimezone(),
+        unit: DurationUnit.Hours,
+    });
+
+    const {minutes} = diffDates({
+        start: date,
+        end: getNowInUserTimezone(),
+        unit: DurationUnit.Minutes,
+    });
+
+    const localeDate = toLocaleString(date, formatPresets.DateFull);
```

Comment:
I think instead of running `toLocaleString` in this function, it should return `undefined` and then `ToniqDateTime` can fallback to using `formatFullDate`, rather than having two different format inputs to `toLocaleString` (in here and in `formatFullDate`).